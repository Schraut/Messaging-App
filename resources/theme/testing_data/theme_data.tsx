import { Theme } from "../theme";

import COLORS from "../colors";
import FONTS from "../fonts";
import { spacing } from "../spacing";

export const THEME_DATA: Theme = {
  colors: COLORS["light"],
  font: FONTS,
  spacing: spacing,
};
