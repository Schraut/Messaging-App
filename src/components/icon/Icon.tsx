import { View, StyleSheet } from "react-native";
import { Theme, useTheme } from "../../../resources/theme/theme";
import {
  MaterialIcons,
  Ionicons,
  FontAwesome5,
  FontAwesome,
  Entypo,
  Octicons,
  Feather,
  AntDesign,
} from "@expo/vector-icons";

interface IconProps {
  iconName: string;
  iconType: string;
  size: number;
  color?: string;
}

const Icon = ({ iconName, iconType, size, color }: IconProps) => {
  const theme = useTheme();
  const styles = stylesHandler(theme);

  // Map the iconType to the correct icon component
  const getIconComponent = (type: string) => {
    switch (type) {
      case "MaterialIcons":
        return MaterialIcons;
      case "Ionicons":
        return Ionicons;
      case "FontAwesome5":
        return FontAwesome5;
      case "FontAwesome":
        return FontAwesome;
      case "Entypo":
        return Entypo;
      case "Octicons":
        return Octicons;
      case "Feather":
        return Feather;
      case "AntDesign":
        return AntDesign;
      default:
        return MaterialIcons; // Default to MaterialIcons if type is not recognized
    }
  };

  // Get the appropriate Icon component based on the iconType
  const IconComponent = getIconComponent(iconType);

  return (
    <View>
      <IconComponent
        name={iconName}
        size={size}
        color={color || theme.colors.staticBlack}
      />
    </View>
  );
};

const stylesHandler = (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.inverseBackground,
      alignItems: "center",
      justifyContent: "center",
    },
  });

export default Icon;
