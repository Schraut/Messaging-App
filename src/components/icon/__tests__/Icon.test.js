import React from "react";
import renderer from "react-test-renderer";
import ThemeContent from "../../../../resources/theme/theme";
import Icon from "../Icon";
import { THEME_DATA } from "../../../../resources/theme/testing_data/theme_data";

const theme = THEME_DATA;

describe("Default Icon", () => {
  it("renders in light mode", () => {
    const tree = renderer
      .create(
        <ThemeContent>
          <Icon
            iconName="add"
            iconType="MaterialIcons"
            size={24}
            color={theme.colors.staticBlack}
          />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    const tree = renderer
      .create(
        <ThemeContent isDarkMode={true}>
          <Icon
            iconName="add"
            iconType="MaterialIcons"
            size={24}
            color={theme.colors.staticBlack}
          />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
