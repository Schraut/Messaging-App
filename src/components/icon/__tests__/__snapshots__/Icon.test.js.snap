// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Icon renders in dark mode 1`] = `
<View>
  <View />
</View>
`;

exports[`Default Icon renders in light mode 1`] = `
<View>
  <View />
</View>
`;
